src/routes/safe/components/Apps/hooks/useAppList.ts

```diff
@@ -130,9 +71,99 @@ const useAppList = (): UseAppListReturnType => {
+    const loadApps = async () => {
+      // recover apps from storage:
+      // * third-party apps added by the user
+      // * disabled status for both static and third-party apps
+      const persistedAppList = (await loadFromStorage<StoredSafeApp[]>(APPS_STORAGE_KEY)) || []
+      let list: (StoredSafeApp & { isDeletable: boolean; networks?: number[] })[] = persistedAppList.map((a) => ({
+        ...a,
+        isDeletable: true,
+      }))
+
+      // merge stored apps with static apps (apps added manually can be deleted by the user)
+      staticAppsList.forEach((staticApp) => {
+        const app = list.find((persistedApp) => persistedApp.url === staticApp.url)
+        if (app) {
+          app.isDeletable = false
+          app.networks = staticApp.networks
+        } else {
+          list.push({ ...staticApp, isDeletable: false })
+        }
+      })
+
+      // filter app by network
+      list = list.filter((app) => {
+        // if the app does not expose supported networks, include them. (backward compatible)
+        if (!app.networks) {
+          return true
+        }
+        return app.networks.includes(getNetworkId())
+      })
+
+      const apps: SafeApp[] = []
+      for (let index = 0; index < list.length; index++) {
+        const currentApp = list[index]
+        const appUrl = currentApp.url.trim()
+
+        const appInfo = {
+          ...getEmptySafeApp(),
+          url: appUrl,
+          name: 'loading',
+          iconUrl: loaderDotsSvg,
+        }
+
+        appInfo.disabled = Boolean(currentApp.disabled)
+        appInfo.isDeletable = Boolean(currentApp.isDeletable) === undefined ? true : currentApp.isDeletable
+
+        apps.push(appInfo)
+      }
+
+      setAppList(apps)
+    }
+
+    if (!appList.length) {
+      loadApps()
+    }
+  }, [appList])
+
+  // fetch each safe-app
+  // replace real data if it was success remove it from the list otherwise
+  useEffect(() => {
+    if (!appList?.length) {
+      return
+    }
+
+    appList
+      .filter((app) => app.loadingStatus === SAFE_APP_LOADING_STATUS.ADDED)
+      .forEach((resApp) => {
+        let cpApps = [...appList]
+        const index = appList.findIndex((currentApp) => currentApp.url === resApp.url)
+        cpApps[index] = { ...cpApps[index], loadingStatus: SAFE_APP_LOADING_STATUS.LOADING }
+        setAppList(cpApps)
+
+        getAppInfoFromUrl(resApp.url).then((res: SafeApp) => {
+          if (res.error) {
+            // if there was an error trying to load the safe-app, remove it from the list
+            cpApps.splice(index, 1)
+            throw Error(`There was a problem trying to load app ${res.url}`)
+          } else {
+            // if the safe app was loaded correctly, update the safe-app info
+            cpApps[index] = res
+          }
+          cpApps = cpApps.sort((a, b) => a.name.toLowerCase().localeCompare(b.name.toLowerCase()))
+          setAppList(cpApps)
+        })
+      })
+  }, [appList])
```

Comment:
Same here

Reply:
it will run until all the requests are resolved. After that, there are not changes in appList.